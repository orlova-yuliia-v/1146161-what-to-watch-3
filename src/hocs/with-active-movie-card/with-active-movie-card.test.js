import React from "react";
import renderer from "react-test-renderer";
import withActiveMovieCard from "./with-active-movie-card.jsx";

const MockComponent = () => <div></div>;
const MockComponentWrapped = withActiveMovieCard(MockComponent);

const movie = [
  {
    title: `Some title`,
    poster: `1.jpg`,
    bgPosterUrl: `https://image-url.com/1.jpg`,
    genre: `Some genre`,
    releaseYear: 2020,
    director: `Director name`,
    actors: [`Actor 1`, `Actor 2`, `Actor 3`, `Actor 4`, `Actor 5`],
    runTime: 2,
    ratingScore: 7.5,
    ratingCount: 1234,
    description: `Film description`,
    previewUrl: `https://preview-url.com/1.mp4`,
    reviews: [
      {
        ratingScore: 8.5,
        date: `September 8, 2019`,
        author: `Yuliia Orlova`,
        text: `Review text`
      }
    ]
  }
];

it(`withActiveCard is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      movie={movie}
      onMovieCardClick={() => { }}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
