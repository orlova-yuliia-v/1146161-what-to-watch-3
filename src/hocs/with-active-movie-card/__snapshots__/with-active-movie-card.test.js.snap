// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveCard is rendered correctly 1`] = `<div />`;
